### ISSUE TITLE
@truffle/contract and @truffle/interface-adapter Critical dependency error

### ISSUE THREAD

**Issue by nogo10** [trufflesuite/truffle#3009]
In my Dapp , I use truffle/contract  without issues until upgraded from truffle-contract to @truffle/contract and web3 v0.20 to v1.2.2 Now get errors (only related to @truffle/contract)

## Steps to Reproduce
```
 "dependencies": {
    "@openzeppelin/gsn-provider": "^0.1.6",
    "@truffle/contract": "^4.2.2",
    "vue": "^2.5.13",
    "vue-router": "^3.0.1",
    "vuex": "^3.0.1",
    "web3": "1.2.2"
  },
```

## Expected Behavior
Should load artifact as usual. Web3 connection works fine without @truffle/contract.

## Actual Results
Compile errors:
```Critical dependency: the request of a dependency is an expression
 warning  in ./node_modules/@truffle/contract/node_modules/web3-eth-accounts/src/scrypt.js```

```Critical dependency: the request of a dependency is an expression
 warning  in ./node_modules/@truffle/interface-adapter/node_modules/web3-eth-accounts/src/scrypt.js```

Web console errors:

`TypeError: Object(...).version is undefined`

```
["./node_modules/readable-stream/lib/_stream_writable.js"]()
["./node_modules/readable-stream/readable-browser.js"]()
["./node_modules/stream-http/lib/response.js"]()
["./node_modules/stream-http/lib/request.js"]()
["./node_modules/stream-http/index.js"]()
["./node_modules/xhr2-cookies/dist/xml-http-request.js"]()
["./node_modules/xhr2-cookies/dist/index.js"]()
["./node_modules/web3-providers-http/src/index.js"]()
["./node_modules/web3-core-requestmanager/src/index.js"]()
["./node_modules/web3-core/src/index.js"]()
["./node_modules/@truffle/contract/node_modules/web3/src/index.js"]()
["./node_modules/@truffle/contract/lib/contract/index.js"]()
["./node_modules/@truffle/contract/index.js"]()
["./node_modules/cache-loader/dist/cjs.js?!./node_modules/babel-loader/lib/index.js!./node_modules/cache-loader/dist/cjs.js?!./node_modules/vue-loader/lib/index.js?!./src/components/SimpleStorage.vue?vue&type=script&lang=js&"]()
```



## Environment
* Operating System: Ubuntu 16.04, Node 10.16.3, Yarn 1.13.0
* Ethereum client: web3 v1.2.2 
stack:
+-- @openzeppelin/cli@2.8.2
| +-- @openzeppelin/upgrades@2.8.0
| | `-- web3@1.2.2  deduped
| +-- truffle-config@1.1.16
| | `-- truffle-provider@0.1.16
| |   +-- truffle-interface-adapter@0.2.5
| |   | `-- web3@1.2.1 
| |   `-- web3@1.2.1 
| +-- typechain-target-web3-v1@1.0.4
| | `-- web3@1.2.7 
| `-- web3@1.2.2  deduped
+-- @openzeppelin/gsn-helpers@0.2.3
| `-- web3@1.2.6 
+-- @openzeppelin/gsn-provider@0.1.10
| `-- web3@1.2.6 
+-- @truffle/contract@4.2.2
| +-- @truffle/interface-adapter@0.4.6
| | `-- web3@1.2.1 
| +-- ethereum-ens@0.8.0
| | `-- web3@1.2.7 
| `-- web3@1.2.1 
`-- web3@1.2.2

**Comment by nogo10** [comment]
Scrypt may not be the issue, here. Instead the error might come from the truffle.js file: Renaming it to truffle-config.js didn't change anything.

**Comment by gnidan** [comment]
Hm, if you use web3 v1.2.1, does the problem persist? We can't upgrade because they've been introducing breaking changes since 1.2.1 (although haven't checked the recent releases)

**Comment by nogo10** [comment]
I have tried with my web3 @ version 1.2.1. Web3 in my Dapp works and gets web3 connection. But when @truffle/contract (with web3 v1.2.1 dep.) is 'required' , web console throws errors.

**Comment by gnidan** [comment]
Can you share those errors please?

**Comment by nogo10** [comment]
as before : running it with the follwing `npm list web3`:



+-- @openzeppelin/cli@2.8.2
| +-- @openzeppelin/upgrades@2.8.0
| | `-- web3@1.2.2 
| +-- truffle-config@1.1.16
| | `-- truffle-provider@0.1.16
| |   +-- truffle-interface-adapter@0.2.5
| |   | `-- web3@1.2.1  deduped
| |   `-- web3@1.2.1  deduped
| +-- typechain-target-web3-v1@1.0.4
| | `-- web3@1.2.7 
| `-- web3@1.2.2 
+-- @openzeppelin/gsn-helpers@0.2.3
| `-- web3@1.2.6 
+-- @openzeppelin/gsn-provider@0.1.10
| `-- web3@1.2.6 
+-- @truffle/contract@4.2.2
| +-- @truffle/interface-adapter@0.4.6
| | `-- web3@1.2.1  deduped
| +-- ethereum-ens@0.8.0
| | `-- web3@1.2.7 
| `-- web3@1.2.1  deduped
`-- web3@1.2.1 


 WARNING  Compiled with 1 warnings                                                                                                                                       5:19:48 PM

 warning  in ./node_modules/web3/node_modules/web3-eth-accounts/src/scrypt.js

Critical dependency: the request of a dependency is an expression

FireFox Browser webconsole error:
`TypeError: Object(...).version is undefined`
...
webpack trace:
```["./node_modules/readable-stream/lib/_stream_writable.js"]()
["./node_modules/readable-stream/readable-browser.js"]()
["./node_modules/stream-http/lib/response.js"]()
["./node_modules/stream-http/lib/request.js"]()
["./node_modules/stream-http/index.js"]()
["./node_modules/xhr2-cookies/dist/xml-http-request.js"]()
["./node_modules/xhr2-cookies/dist/index.js"]()
["./node_modules/web3-providers-http/src/index.js"]()
["./node_modules/web3-core-requestmanager/src/index.js"]()
["./node_modules/web3-core/src/index.js"]()
["./node_modules/web3/src/index.js"]()
["./node_modules/cache-loader/dist/cjs.js?!./node_modules/babel-loader/lib/index.js!./node_modules/cache-loader/dist/cjs.js?!./node_modules/vue-loader/lib/index.js?!./src/components/SimpleStorage.vue?vue&type=script&lang=js&"]()
["./src/components/SimpleStorage.vue?vue&type=script&lang=js&"]() ```

**Comment by nogo10** [comment]
I got past that error (```TypeError: Object(...).version is undefined```) by 
removing  @truffle/contract from my package.json then upgrading web3 from v1.2.1 to v1.2.2 and importing directly the dist/web3..js compiled web3 into my Dapp.  ```import Web3 from 'web3/dist/web3.min.js'```

Until II get an answer/solution I must get my contract instance with web3.eth.Contract web3 method only (after extracting the ABI data from the artifact )

**Comment by nogo10** [comment]
BTW I did retry to import @truffle/contract once I got web3 upgraded and working; the result was the same webconsole error cited above: ```TypeError: Object(...).version is undefined```

**Comment by eggplantzzz** [comment]
@nogo10 So you are getting these errors while trying to compile? Can you please give me some specific steps I can take to generate your errors?

**Comment by nogo10** [comment]
Well unbox this truffle box (https://github.com/standup75/vue-box) or my version (https://gitlab.com/chevronx/vue-box-gsn)
'upgrade' the web3 to 1.2.x (any version) then yarn install , compile deploy to ganache, yarn serve. All on Node version 10.16.3.

**Comment by nogo10** [comment]
of course upgrading truffle-contract to @truffle/contract too

**Comment by eggplantzzz** [comment]
Hmm, I'm not getting the same error as you. I updated to `@truffle/contract` 4.2.2 (4.2.3 has a bug so don't use that one) and `web3` 1.2.1. I get a console error about not being able to call `slice` of `undefined`. I'm not sure what is up. I'll try the other project.

**Comment by eggplantzzz** [comment]
Yeah I'm getting the same error with your project when I change to use `@truffle/contract`.
`Cannot read property 'slice' of undefined` in the webpack build.
I'm not sure, maybe we'll have to take a simpler project and try to webpack `@truffle/contract` to see how it works with webpack.

**Comment by nogo10** [comment]
@nicosampler was this issue resolved in your code?

**Comment by nicosampler** [comment]
> @nicosampler was this issue resolved in your code?

Nop, I had to silent the error in webpack to make the build complete in the CI.

**Comment by dvargas92495** [comment]
What's the status here? I'm hitting this issue with a CRA and trying to build on CI, so can't just ignore on webpack. Is there anything blocking upgrading web3 to latest within `@truffle/interface-adapter` ?
